pkg/controller/ephemeralcluster/reconciler.go

```diff
@@ -300,7 +384,21 @@ func (r *reconciler) createProwJob(ctx context.Context, log *logrus.Entry, ec *e
 		upsertCondition(&ec.Status, ephemeralclusterv1.ProwJobCreating, status, r.now(), reason, msg)
 	}
 
-	ciOperatorConfig, err := r.generateCIOperatorConfig(log, ec)
+	annotations := ec.Annotations
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+
+	prMeta, err := parsePRMeta(annotations[PREventPayload], annotations[PREventHeaders])
+	if err != nil {
+		log.WithError(err).Error("parse pull request meta")
+		err = fmt.Errorf("parse pull request meta: %w", err)
```

Comment:
Here you are mutating the existing err.. In the end you will return the wrong error... Let's just put this in the global const of errors to ensure consistency.

Reply:
Here this is what I'm doing:
```golang
result, err := foobar(...)
if err != nil {
    log.WithError(err).Error("foobar failed")
    err = fmt.Errorf("foobar failed: %w", err)
    return err
}
```
The `err` I return is the original one plus some context about the `foobar` function that has failed. Not returning the original error is intentional because I want the extra information to be there.